### PR TITLE
support screenshot for safari and web view app

### PR DESCRIPTION
**Pull request by yaphatak**

### CODE REVIEW

File: lib/driver.js [hunk 1]
```diff
@@ -129,7 +130,10 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.launch(sessionId);
 
       this.proxyReqRes = this.wda.proxyReqRes.bind(this.wda);
-      this.jwpProxyActive = true;
+      // if we keep proxy active then all call will route to proxy
```
Comment (yaphatak) [inline lib/driver.js]: I didn't like this conditional thing but if we don't do it then all calls will go to XCUITestDriver instead of safari. We need to change setContext later to support Native functionality for safari browser (lower pro)

File: lib/commands/screenshot.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+let commands = {};
```
Comment (imurchie) [inline lib/commands/screenshot.js]: Why is this file necessary? Can we not just proxy _all_ `screenshot` requests to WDA?

Comment (imurchie) [inline lib/commands/screenshot.js]: What if, rather than stopping proxying and doing everything manually (because every command that uses WDA is going to have to be duplicated in this scheme), we make use of the `appium-safari-driver` NO_PROXY list and whatnot. That is, when we are in a Safari/Webview context, calls to `proxyActive`, `canProxy`, and `getProxyAvoidList` defer to SafariDriver?

Then we would have something like 

```js
class XCUITestDriver extends BaseDriver {
  // ...

  getProxyAvoidList () {
    if (this.inWebview()) {
      return this._safariDriver.getProxyAvoidList();
    } else {
      // do what we now do
    }

  // ...
}
```

This is a bit more work in the short term, but to my mind it keeps the logic more close to the logic of a normal driver. The separation between the delegated driver and the normal driver is less.

But I'm perfectly happy to entertain dissenting opinions! :)

**Comment by imurchie** [comment]
_[Note: I moved this out of the in-line comment so that the discussion wouldn't be obscured if the commit changed]_

What if, rather than stopping proxying and doing everything manually (because every command that uses WDA is going to have to be duplicated in this scheme), we make use of the `appium-safari-driver` NO_PROXY list and whatnot. That is, when we are in a Safari/Webview context, calls to `proxyActive`, `canProxy`, and `getProxyAvoidList` defer to SafariDriver?

Then we would have something like 

```js
class XCUITestDriver extends BaseDriver {
  // ...

  getProxyAvoidList () {
    if (this.inWebview()) {
      return this._safariDriver.getProxyAvoidList();
    } else {
      // do what we now do
    }

  // ...
}
```

This is a bit more work in the short term, but to my mind it keeps the logic more close to the logic of a normal driver. The separation between the delegated driver and the normal driver is less.

But I'm perfectly happy to entertain dissenting opinions! :)

**Comment by yaphatak** [comment]
1. This is only for screenshot. For all other actions user should switch to Native_App and then do least of actions if they have to use WDA. But one can argue that we still need to support few actions in this manner eg deactivate_app
2. I would like change you suggested but I still think it won't work. Here we need to actually do proxy and not avoid proxy. So it does feel like we need some similar kinda list where we force proxy

**Comment by imurchie** [comment]
(1) This will be for anything that we don't need to re-implement in Safari, no? Any tasks that are not web related.

(2) The list exported from `getProxyAvoidList` is the things that the driver _can_ do. So it would not have `screenshot` in there, and the proxying mechanism would automatically do the proxying. So for SafariDriver the list would include things like getting a url, etc.

**Comment by yaphatak** [comment]
I think that make sense. Let me close this PR. List will include most of calls like find element, click..in fact most of calls
